src/vcastTestInterface.ts

```diff
@@ -353,27 +353,59 @@ export async function getResultFileForTest(testID: string) {
   // This function will return the path to the result file if it is already saved
   // in the globalTestStatus array, otherwise it will ask Python to generate the report
   let resultFile: string = globalTestStatusArray[testID].resultFilePath;
-  if (!fs.existsSync(resultFile)) {
-    let enviroPath = getEnviroPathFromID(testID);
 
+  // Check if the file already exists
+  if (!fs.existsSync(resultFile)) {
+    // Generate the environment path and request the test report from Python
+    const enviroPath = getEnviroPathFromID(testID);
     const commandStatus = await getTestExecutionReport(enviroPath, testID);
 
-    if (commandStatus.errorCode == 0) {
+    // Check if report generation was successful
+    if (commandStatus.errorCode === 0) {
       const firstLineOfOutput: string = commandStatus.stdout
         .split("\n", 1)[0]
         .trim();
-      resultFile = firstLineOfOutput.replace("REPORT:", "");
 
-      if (!fs.existsSync(resultFile)) {
+      // Handle the case where the output contains "REPORT"
+      if (firstLineOfOutput.includes("REPORT:")) {
+        // Verify if the generated report file actually exists
+        if (!fs.existsSync(resultFile)) {
+          vscode.window.showWarningMessage(
+            `The Report: ${resultFile} does not exist.`
+          );
+          vectorMessage(`The Report: ${resultFile} does not exist.`);
+        } else {
+          // This is the normal case --> delete the REPORT to only have the file name
+          resultFile = firstLineOfOutput.replace("REPORT:", "");
+        }
+      }
+      // If the first line of output contains "Error" --> Test result generation failed
+      else if (firstLineOfOutput.includes("Error:")) {
+        const errorDetails = firstLineOfOutput.split("Error:")[1].trim();
         vscode.window.showWarningMessage(
-          `Results report: '${resultFile}' does not exist`
+          `Execution report was not successfully generated. Error details: \n${errorDetails}`
+        );
+        vectorMessage(
+          `Execution report was not successfully generated. Error details: \n${errorDetails}`
         );
-        vectorMessage(`Results report: '${resultFile}' does not exist`);
-        vectorMessage(commandStatus.stdout, errorLevel.info, indentString);
       }
-
-      globalTestStatusArray[testID].resultFilePath = resultFile;
+      // Handle other unexpected cases (After successfull test generation, but without the "REPORT:" string)
+      else {
+        vscode.window.showWarningMessage(
+          `Unexpected Error: \n${commandStatus.stdout}`
+        );
+        vectorMessage(`Unexpected Error: \n${commandStatus.stdout}`);
```

Comment:
Let's make this string only once

Reply:
The string is once in the Output pane (which makes it easier for testing and logging I think) and the other one is the warning message popup for the user. What do you want me to delete ?

Reply:
No, no ...

```ts
const errorStr = `Unexpected Error: \n${commandStatus.stdout}`;
vscode.window.showWarningMessage(errorStr);
vectorMessage(errorStr);
```